lib/features/home/presentation/controller/ideal_type_notifier.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:atwoz_app/app/constants/enum.dart';
+import 'package:atwoz_app/features/home/data/data.dart';
+import 'package:atwoz_app/features/home/domain/domain.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'ideal_type_notifier.g.dart'; // 코드 생성을 위한 부분
+
+final idealTypeRepository = Provider<IdealTypeRepository>((ref) {
+  return IdealTypeRepositoryImpl();
+});
+
+@riverpod
+class IdealTypeNotifier extends AutoDisposeAsyncNotifier<IdealType> {
+  @override
+  Future<IdealType> build() async {
+    final repository = ref.read(idealTypeRepository);
+    final idealType = await repository.getIdealType();
+    return idealType;
+  }
+
+  void updateAgeRange(int minAge, int maxAge) {
+    if (state.hasValue) {
+      final updatedIdealType = state.value!.copyWith(
+        minAge: minAge,
+        maxAge: maxAge,
+      );
+      state = AsyncData(updatedIdealType);
+    }
```

Comment:
```suggestion
    if(!state.hasValue) return;
    state = AsyncData(state.requireValue.copyWith(
        minAge: minAge,
        maxAge: maxAge,
    ));
```
위와 같은 구조를 사용하면 code depth를 줄이고, Non-null assertion operator
연산자를 사용하지 않아도 될 것 같아요~ 
(내부적으로 동일하지만, 프로젝트내 강제 캐스팅을 줄이면 좋을 것 같아요)

아래도 동일합니다~